pkg/genericapiserver/genericapiserver.go

```diff
@@ -547,3 +553,40 @@ func waitForSuccessfulDial(https bool, network, address string, timeout, interva
 	}
 	return err
 }
+
+// getNamedCertificateMap returns a map of strings to *tls.Certificate, suitable for use in
+// tls.Config#NamedCertificates. Returns an error if any of the certs cannot be loaded.
+// Returns nil if len(namedKeyCerts) == 0
+func getNamedCertificateMap(namedKeyCerts []NamedKeyCert) (map[string]*tls.Certificate, error) {
+	if len(namedKeyCerts) == 0 {
+		return nil, nil
+	}
+	namedCerts := map[string]*tls.Certificate{}
+	for _, nkc := range namedKeyCerts {
+		cert, err := tls.LoadX509KeyPair(nkc.CertFile, nkc.KeyFile)
+		if err != nil {
+			return nil, err
+		}
+		if len(nkc.Names) > 0 {
+			for _, name := range nkc.Names {
```

Comment:
I think we should give auto-generation precedence, but enforce that the first cert given trumps any later one. That matches your example. WDYT?